eng/scripts/docs/Docs-ToC.ps1

```diff
@@ -0,0 +1,154 @@
+        $packageLevelReadmeName = $readmeMetadata.DocsMsReadMeName
+    }
+
+    $packageTocHeader = $packageMetadata.Package
+    if ($packageMetadata.DisplayName) {
+        $packageTocHeader = $packageMetadata.DisplayName
+    }
+
+    $version = ""
+    $isPreview = $false
+    if($packageMetadata.VersionGA -ne "") {
+        $version = $packageMetadata.VersionGA
+    }
+    else {
+        $version = $packageMetadata.VersionPreview
+        $isPreview = $true
+    }
+    $children = Get-Toc-Children -package $packageMetadata.Package -groupId $packageMetadata.GroupId `
+        -version $version -docRepoLocation $docRepoLocation -isPreview $isPreview
+
+    if (!$children) {
+        Write-Host "Did not find the package namespaces for $($packageMetadata.GroupId):$($packageMetadata.Package):$version"
+    }
+    $output = [PSCustomObject]@{
+        PackageLevelReadmeHref = "~/docs-ref-services/{moniker}/$packageLevelReadmeName-readme.md"
+        PackageTocHeader       = $packageTocHeader
+        TocChildren            = $children
+    }
+    return $output
+}
+
+function Get-java-DocsMsTocChildrenForManagementPackages($packageMetadata, $docRepoLocation) {
+    $children = @()
+    foreach ($package in $packageMetadata) {
+        $version = $package.VersionGA
+        $isPreview = $false
+        if($package.VersionPreview) {
+            $version = $package.VersionPreview
+            $isPreview = $true
+        }
+        $packageToc = Get-Toc-Children -package $package.Package -groupId $package.GroupId -version $version `
+            -docRepoLocation $docRepoLocation -isPreview $isPreview
+        $children += $packageToc
+    }
+    # Flatten the children if multiple packages.
+    return @($children)
+}
+
+# This is a helper function which fetch the java package namespaces from javadoc jar.
+# Here are the major workflow:
+# 1. Read the ${package}.txt under /metadata folder
+# 2. If file not found, then download javadoc jar from maven repository.
+# 3. If there is 'element-list' in javadoc jar, then copy to destination ${package}.txt.
+# 4. If no 'element-list', then parse the 'overview-frame.html' for the namespaces and copy to destination.
+# 5. If no 'overview-frame.html', then read folder 'com/azure/...' for namespaces. E.g some mgmt packages use this structure.
+# 6. Otherwise, return empty children.
+function Get-Toc-Children($package, $groupId, $version, $docRepoLocation, $isPreview) {
+    # Looking for the txt
+    $folder = "latest" 
+    if ($isPreview) {
+        $folder = "preview"
+    } 
+    $filePath = Join-Path "$docRepoLocation/metadata/$folder" "$package.txt"
+    if (!(Test-Path $filePath)) {
+        # Download from maven
+        # javadoc jar url. e.g.: https://repo1.maven.org/maven2/com/azure/azure-core/1.25.0/azure-core-1.25.0-javadoc.jar
+        $groupId = "$groupId" -replace "\.", "/"
+        $mavenJavadocJarUrl = "$packageDownloadUrl/$groupId/$package/$version/$package-$version-javadoc.jar"
```

Comment:
Since we could be using the daily package feed in the future we should use `mvn` to fetch the javadoc jar instead of inferring the URL: 

```
mvn dependency:copy -Dartifact=com.azure:azure-core:1.26.0:javadoc -DoutputDirectory=.
```

https://maven.apache.org/plugins/maven-dependency-plugin/copy-mojo.html